### ISSUE TITLE
Permission Error Handling Page

### ISSUE THREAD

**Issue by wreckage0907** [rtCamp/next-pms#2077]
# 🚀 Feature Request

## **Description**
Basically when a developer who doesnt have permission to hit the URL an empty page layout without any data as the backend APIs have strict permission handling , but by opening this page they can still access the components and simply overload the system with empty queries . instead could display a page saying not accessible for your permission so it clearly blocks them 

<img width="1920" height="978" alt="Image" src="https://github.com/user-attachments/assets/c1b11b39-52f2-4063-8880-f83614bfc99e" />

## **Motivation**
- for user accessibility and better quality in terms of app 

## **Acceptance Criteria:**
- [] when the API returns httpStatus === 403 / exc_type === "PermissionError" render a page that says not accessible 

## **Additional Context**
[Add any other context, links, or references that would help explain this feature request.]

**Comment by renishsurani** [comment]
When I try to open pages that employees aren't allowed to see, I now see the 404 page.

<img width="1468" height="869" alt="Image" src="https://github.com/user-attachments/assets/536be642-4599-430d-a8a2-537ccfbc7d8d" />

cc: @Ajay-Jaiswal416

**Comment by renishsurani** [comment]
I have verified that, as an employee, I can see the proper 404 page for which I do not have access to open.

cc: @Ajay-Jaiswal416
